### ISSUE TITLE
[BUG]: load_checkpoint error

### ISSUE THREAD

**Issue by bobo0810** [hpcaitech/ColossalAI#3886]
### 🐛 Describe the bug

The weight of data parallel（DP） and pipeline parallel （PP） saved could not be loaded. Report the following error

save_checkpoint("xxx.pt", epoch=cfg.epochs, model=engine.model)
![image](https://github.com/hpcaitech/ColossalAI/assets/26057879/710eed4b-dfa2-4fb3-aaed-f999f7d238bf)

### Environment

absl-py                           0.12.0
addict                            2.4.0
aiohttp                           3.8.1
aiosignal                         1.2.0
alabaster                         0.7.12
albumentations                    1.1.0
alluxio                           0.1.3
anyio                             3.5.0
apache-beam                       2.27.0
apache-flink                      1.14.4
apache-flink-libraries            1.14.4
apex                              0.1
appdirs                           1.4.4
argon2-cffi                       20.1.0
arsenal-magic-prod-dev            2.2.70
arviz                             0.12.0
ascii-graph                       1.5.1
astor                             0.8.1
async-generator                   1.10
async-timeout                     4.0.2
attrs                             20.3.0
audioread                         2.1.9
autovizwidget                     0.18.0
avro-python3                      1.9.2.1
Babel                             2.9.0
backcall                          0.2.0
bce-python-sdk                    0.8.64
beautifulsoup4                    4.9.3
bert4keras                        0.11.3
bleach                            3.3.0
blis                              0.7.4
blosc                             1.10.6
bokeh                             2.4.2
boto3                             1.17.32
botocore                          1.20.32
brotlipy                          0.7.0
cachetools                        5.0.0
catalogue                         1.0.0
certifi                           2020.12.5
cffi                              1.14.3
cfgv                              3.3.1
cftime                            1.6.0
chardet                           3.0.4
charset-normalizer                2.0.12
chinesecalendar                   1.7.2
click                             8.1.3
cloudpickle                       1.2.2
cmake                             3.26.3
codecov                           2.1.11
colorama                          0.4.4
conda                             4.9.2
conda-build                       3.21.4
conda-pack                        0.6.0
conda-package-handling            1.7.2
configparser                      5.2.0
confluent-kafka                   1.8.2
convertdate                       2.4.0
coverage                          5.5
cramjam                           2.5.0
crcmod                            1.7
cryptography                      3.2.1
cxxfilt                           0.2.2
cycler                            0.10.0
cymem                             2.0.5
Cython                            0.28.4
dask                              2022.5.0
DataProperty                      0.50.0
dataset                           0.1.3
datasets                          2.11.0
debugpy                           1.6.0
decorator                         4.4.2
deepspeed                         0.5.10
defusedxml                        0.7.1
demjson                           2.2.4
deprecat                          2.1.1
Deprecated                        1.2.13
diffusers                         0.12.1
dill                              0.3.6
distlib                           0.3.4
distributed                       2022.5.0
DLLogger                          0.1.0
dnspython                         2.2.1
docopt                            0.6.2
docutils                          0.16
easydict                          1.9
entrypoints                       0.3
ephem                             4.1.3
fastavro                          0.23.6
fastjsonschema                    2.15.3
FastNLP                           0.7.0
fastparquet                       0.8.1
fastprogress                      1.0.2
fasttext                          0.9.2
filelock                          3.6.0
fire                              0.4.0
flake8                            3.7.9
Flask                             2.1.2
Flask-Babel                       2.0.0
frozenlist                        1.3.0
fsspec                            2022.3.0
funcy                             1.17
future                            0.18.2
gensim                            4.2.0
gitdb                             4.0.9
GitPython                         3.1.27
glob2                             0.7
google-auth                       2.6.6
google-auth-oauthlib              0.5.1
google-pasta                      0.2.0
graphsurgeon                      0.4.5
graphviz                          0.20
grpcio                            1.54.0
grpcio-tools                      1.45.0
gssapi                            1.7.3
h5py                              3.2.1
hanziconv                         0.3.2
hdfs                              2.7.0
hdijupyterutils                   0.18.0
HeapDict                          1.0.1
hijri-converter                   2.2.3
hjson                             3.1.0
holidays                          0.13
html2text                         2020.1.16
htmlmin                           0.1.12
httplib2                          0.17.4
huggingface-hub                   0.13.4
hypothesis                        4.50.8
identify                          2.5.0
idna                              2.10
ImageHash                         4.2.1
imageio                           2.9.0
imagesize                         1.2.0
imbalanced-learn                  0.9.0
imblearn                          0.0
importlib-metadata                3.7.3
inflect                           5.3.0
influxdb                          5.3.1
iniconfig                         1.1.1
interval                          1.0.0
ipdb                              0.13.7
ipykernel                         5.5.0
ipython                           7.21.0
ipython-genutils                  0.2.0
IPythonDisplayFireworks           0.0.4
IPythonDisplayTurtle              0.0.33
ipywidgets                        7.7.0
itsdangerous                      2.1.2
jedi                              0.17.0
jieba                             0.42.1
jieba-fast                        0.53
Jinja2                            3.1.2
jmespath                          0.10.0
joblib                            1.1.0
json5                             0.9.5
jsonschema                        3.0.2
jupyter-client                    6.1.12
jupyter-contrib-core              0.3.3
jupyter-contrib-nbextensions      0.5.1
jupyter-core                      4.7.1
jupyter-highlight-selected-word   0.2.0
jupyter-latex-envs                1.4.6
jupyter-nbextensions-configurator 0.4.1
jupyter-server                    1.17.0
jupyter-server-proxy              3.2.1
jupyter-tensorboard               0.2.0
jupyterlab                        2.3.1
jupyterlab-git                    0.24.0
jupyterlab-nvdashboard            0.5.0
jupyterlab-pygments               0.1.2
jupyterlab-server                 1.2.0
jupyterlab-widgets                1.1.0
jupytext                          1.11.0
kazoo                             2.5.0
Keras                             2.3.1
Keras-Applications                1.0.8
keras-bert                        0.89.0
keras-embed-sim                   0.10.0
keras-layer-normalization         0.16.0
keras-multi-head                  0.29.0
keras-pos-embd                    0.13.0
keras-position-wise-feed-forward  0.8.0
Keras-Preprocessing               1.1.2
keras-self-attention              0.51.0
keras-transformer                 0.40.0
kiwisolver                        1.3.1
korean-lunar-calendar             0.2.1
krb5                              0.3.0
kubernetes                        17.17.0
libarchive-c                      2.9
librosa                           0.8.0
lightgbm                          3.3.2
lightning-lite                    1.8.0
lightning-utilities               0.3.0
lit                               16.0.1
llvmlite                          0.38.0
lmdb                              1.1.1
locket                            1.0.0
log-tool                          1.1.9
LunarCalendar                     0.0.9
lxml                              4.8.0
lz4                               4.0.0
Mako                              1.1.4
Markdown                          3.3.4
markdown-it-py                    0.6.2
MarkupSafe                        2.1.1
maskrcnn-benchmark                0.1
matplotlib                        3.3.4
matrixprofile                     1.1.10
mbstrdecoder                      1.0.1
mccabe                            0.6.1
mdit-py-plugins                   0.2.6
minio                             5.0.10
missingno                         0.5.1
mistune                           0.8.4
mlperf-compliance                 0.0.10
mmcv-full                         1.5.0
mock                              2.0.0
mpmath                            1.3.0
msgfy                             0.1.0
msgpack                           1.0.3
multidict                         6.0.2
multimethod                       1.8
multiprocess                      0.70.14
murmurhash                        1.0.5
mypy-extensions                   1.0.0
nbclient                          0.5.3
nbconvert                         6.5.0
nbdime                            2.1.1
nbformat                          5.4.0
nest-asyncio                      1.5.1
netCDF4                           1.5.8
networkx                          2.8
ninja                             1.11.1
nltk                              3.5
nodeenv                           1.6.0
notebook                          6.2.0
numba                             0.55.1
numexpr                           2.8.1
numpy                             1.21.0
nvidia-cublas-cu11                11.10.3.66
nvidia-cuda-cupti-cu11            11.7.101
nvidia-cuda-nvrtc-cu11            11.7.99
nvidia-cuda-runtime-cu11          11.7.99
nvidia-cudnn-cu11                 8.5.0.96
nvidia-cufft-cu11                 10.9.0.58
nvidia-curand-cu11                10.2.10.91
nvidia-cusolver-cu11              11.4.0.1
nvidia-cusparse-cu11              11.7.4.91
nvidia-dali-cuda110               0.31.0
nvidia-dlprof-pytorch-nvtx        1.0.0
nvidia-nccl-cu11                  2.14.3
nvidia-nvtx-cu11                  11.7.91
nvidia-pyprof                     3.9.0
nvidia-tensorboard                1.15.0+nv21.3
nvidia-tensorboard-plugin-dlprof  0.12
oauth2client                      4.1.3
oauthlib                          3.2.0
onnx                              1.8.0
onnxruntime                       1.7.0
opencv-python                     4.5.5.64
opt-einsum                        3.3.0
packaging                         21.3
paddlepaddle-gpu                  2.2.2.post112
pandas                            1.1.5
pandas-profiling                  3.2.0
pandocfilters                     1.4.3
parso                             0.8.1
partd                             1.2.0
pathvalidate                      2.3.2
patsy                             0.5.2
pbr                               5.9.0
pexpect                           4.8.0
phik                              0.12.2
pickleshare                       0.7.5
Pillow                            9.1.0
Pillow-SIMD                       7.0.0.post3
pip                               23.1.2
pipdeptree                        2.2.1
pkginfo                           1.7.0
plac                              1.1.0
platformdirs                      2.5.2
plotly                            5.7.0
pluggy                            0.13.1
polygraphy                        0.25.1
pooch                             1.3.0
pre-commit                        2.19.0
preshed                           3.0.2
prettytable                       2.1.0
progressbar                       2.5
prometheus-client                 0.9.0
prompt-toolkit                    3.0.8
protobuf                          3.19.0
psutil                            5.8.0
ptyprocess                        0.7.0
py                                1.10.0
py-cpuinfo                        9.0.0
py4j                              0.10.9.3
pyarrow                           11.0.0
pyasn1                            0.4.8
pyasn1-modules                    0.2.8
pybind11                          2.6.2
pyclipper                         1.3.0.post2
pycocotools                       2.0+nv0.4.0
pycodestyle                       2.5.0
pycosat                           0.6.3
pycparser                         2.20
pycryptodome                      3.14.1
pycuda                            2020.1
pydantic                          1.9.0
pydot                             1.4.2
pyecharts                         1.9.1
pyflakes                          2.1.1
Pygments                          2.8.1
pyhelm                            2.14.5
pykafka                           2.8.0
pyLDAvis                          3.3.1
pymc3                             3.11.5
PyMeeus                           0.5.11
pymongo                           3.12.3
PyMySQL                           1.0.2
pynvml                            8.0.4
pyOpenSSL                         19.1.0
pyparsing                         2.4.7
pyre-extensions                   0.0.23
pyrsistent                        0.17.3
PySocks                           1.7.1
pyspark                           3.2.1
pyspnego                          0.5.2
pytablewriter                     0.47.0
pytest                            6.2.2
pytest-cov                        2.11.1
pytest-pythonpath                 0.7.3
python-dateutil                   2.8.0
python-hostlist                   1.21
python-nvd3                       0.15.0
python-slugify                    4.0.1
python-snappy                     0.6.1
pytools                           2021.2
pytorch-lightning                 1.8.0
pytorch-quantization              2.1.0
pytorch-transformers              1.1.0
pytz                              2021.1
PyWavelets                        1.1.1
PyYAML                            5.4.1
pyzmq                             22.0.3
qudida                            0.0.4
redis                             4.2.2
regex                             2021.3.17
requests                          2.27.1
requests-kerberos                 0.14.0
requests-oauthlib                 1.3.1
resampy                           0.2.2
responses                         0.18.0
revtok                            0.0.3
rsa                               4.8
ruamel_yaml                       0.15.87
s3transfer                        0.3.6
sacrebleu                         1.2.10
sacremoses                        0.0.35
scikit-image                      0.19.2
scikit-learn                      1.0.2
scipy                             1.7.3
seaborn                           0.11.2
semver                            2.13.0
Send2Trash                        1.5.0
sentencepiece                     0.1.95
setuptools                        57.5.0
shap                              0.40.0
Shapely                           1.8.2
shellcheck-py                     0.8.0.4
simpervisor                       0.4
simplejson                        3.17.6
six                               1.15.0
sklearn                           0.0
slicer                            0.0.7
smart-open                        6.0.0
smmap                             5.0.0
sniffio                           1.2.0
snowballstemmer                   2.1.0
sortedcontainers                  2.4.0
SoundFile                         0.10.3.post1
soupsieve                         2.2
sox                               1.4.1
spacy                             2.3.5
sparkmagic                        0.18.0
Sphinx                            3.5.2
sphinx-glpi-theme                 0.3
sphinx-rtd-theme                  0.5.1
sphinxcontrib-applehelp           1.0.2
sphinxcontrib-devhelp             1.0.2
sphinxcontrib-htmlhelp            1.0.3
sphinxcontrib-jsmath              1.0.1
sphinxcontrib-qthelp              1.0.3
sphinxcontrib-serializinghtml     1.1.4
srsly                             1.0.5
statsmodels                       0.13.2
stumpy                            1.11.1
subword-nmt                       0.3.3
supermutes                        0.2.5
sympy                             1.11.1
tabledata                         1.1.3
tabulate                          0.8.9
tangled-up-in-unicode             0.2.0
tblib                             1.7.0
tenacity                          8.0.1
tensorboard                       2.12.2
tensorboard-data-server           0.7.0
tensorboard-plugin-wit            1.8.1
tensorrt                          7.2.2.3
tensorrtserver                    1.9.0
termcolor                         1.1.0
terminado                         0.9.3
terminaltables                    3.1.10
testpath                          0.4.4
text-unidecode                    1.3
Theano-PyMC                       1.1.2
thinc                             7.4.5
threadpoolctl                     2.1.0
tifffile                          2022.5.4
tinycss2                          1.1.1
tokenizers                        0.12.1
toml                              0.10.2
toolz                             0.11.2
torch                             2.0.0
torchmetrics                      0.11.4
torchtext                         0.9.0a0
torchvision                       0.9.0a0
tornado                           6.1
tqdm                              4.51.0
traitlets                         5.1.1
transformers                      4.18.0
triton                            2.0.0
tsfresh                           0.19.0
typeguard                         2.13.3
typepy                            1.1.4
typing_extensions                 4.5.0
typing-inspect                    0.8.0
uff                               0.6.9
Unidecode                         1.2.0
urllib3                           1.25.11
virtualenv                        20.14.1
visions                           0.7.4
visualdl                          2.2.3
wasabi                            0.8.2
wcwidth                           0.2.5
webencodings                      0.5.1
websocket-client                  1.3.2
Werkzeug                          2.1.2
wheel                             0.35.1
widgetsnbextension                3.6.0
wordcloud                         1.8.1
wrapt                             1.10.11
xarray                            2022.3.0
xeus-python                       0.13.7
xeus-python-shell                 0.2.0
xformers                          0.0.18
xgboost                           1.6.0
xxhash                            3.2.0
yacs                              0.1.8
yapf                              0.32.0
yarl                              1.7.2
zict                              2.2.0
zipp                              3.4.1
zstd                              1.4.9.1

**Comment by jiangmingyan** [comment]
you can refer to our new checkpoints api
. https://colossalai.org/docs/basics/booster_checkpoint
